File: Cpp/Unit01/Exercise09/src/exercise09.cpp
#include<iostream>
#include<string>
#include<fstream>

using namespace std;

int main(int argc, char** argv) {
    // Getting the path to the Input/Output folders
    string fs(argv[1]);
    string inputPath = "../Unit01/Exercise09/Inputs/" + fs;
    string outputPath = "../Unit01/Exercise09/Outputs/output.txt";

    // Getting the number from the input file
    float n;
    ifstream inFile;
    inFile.open(inputPath);
    inFile >> n;
    inFile.close();

    // Writing out the cube of the number
    ofstream outFile;
    outFile.open(outputPath);
    outFile << n*n*n;
    outFile.close();

    return 0;
}

File: Fortran90/Unit01/Exercise09/src/exercise09.f90
program exercise09
    implicit none

    character(len=120) :: inputPath, outputPath, fileName
    real :: a

    ! Getting the input file name and output file
    call get_command_argument(1, fileName)
    inputPath = "../Unit01/Exercise09/Inputs/"//trim(fileName)
    outputPath = "../Unit01/Exercise09/Outputs/output.txt"

    ! Reading the number in the file
    open(1, file=inputPath)
    read(1,*) a
    close(1)

    open(1, file=outputPath)
    write(1,*) a*a*a
    close(1)

end program exercise09